src/middlewared/middlewared/plugins/truecommand/__init__.py

```diff
@@ -10,7 +10,11 @@ async def _event_system_ready(middleware, event_type, args):
 
 async def setup(middleware):
     await middleware.call('truecommand.config')
-    middleware.event_register('truecommand.config', 'Sent on TrueCommand configuration changes.')
+    middleware.event_register(
+        'truecommand.config',
+        'Sent on TrueCommand configuration changes.',
+        roles=['READONLY_ADMIN']
```

Comment:
Should we perhaps add role for truecommand as well and then use it's read role here ?

Reply:
Yeah, I'm doing the roles based changes in a separate PR because we have to fix some other things.